multi_imbalance/utils/data.py

```diff
@@ -20,3 +31,48 @@ def construct_flat_2pc_df(X, y) -> pd.DataFrame:
     df = pd.concat([X_df, y], axis=1)
 
     return df
+
+
+def get_project_root() -> Path:
+    """Returns project root folder."""
+    return Path(__file__).parent.parent.parent
+
+
+def preprocess_dataset(path):
+    data, meta = arff.loadarff(path)
+
+    df = pd.DataFrame(data)
+    y_index = len(df.columns) - 1
+    y = df.pop(df.columns[y_index])
+
+    le = LabelEncoder()
+    y = le.fit_transform(y)
+
+    categorical_feature_mask = df.dtypes == object
+
+    categorical_cols = df.columns[categorical_feature_mask].tolist()
+    non_categorical_cols = df.columns[~categorical_feature_mask].tolist()
+
+    df[categorical_cols] = df[categorical_cols].replace({b'?': np.NaN})
+
+    mode = df.mode().iloc[0]
+    mean = df.filter(non_categorical_cols).mean()
+
+    df[categorical_cols] = df.filter(categorical_cols).fillna(mode)
+    df[non_categorical_cols] = df.filter(non_categorical_cols).fillna(mean)
+
+    X = pd.get_dummies(df, columns=categorical_cols)
+    return X.to_numpy(), y
+
+
+def load_arff_datasets():
+    dataset_paths = glob.glob(f'{get_project_root()}/data/arff/*')
```

Comment:
we could rewrite this somehow in order not to enforce /arff already to exist. Maybe add arff dataset in .zip or .tar.gz analogously to our datasets/_data_loader.py file